test/conformance/ingress/util.go

```diff
@@ -200,14 +212,112 @@ func CreateIngressReady(t *testing.T, clients *test.Clients, spec v1alpha1.Ingre
 		t.Fatalf("Error getting Ingress: %v", err)
 	}
 
+	var tlsConfig *tls.Config
+	if len(ing.Spec.TLS) > 0 {
+		// CAs are added to this as TLS secrets are created.
+		tlsConfig = &tls.Config{
+			RootCAs: rootCAs,
+		}
+	}
+
 	// Create a client with a dialer based on the Ingress' public load balancer.
 	return name, &http.Client{
 		Transport: &http.Transport{
-			DialContext: CreateDialContext(t, ing, clients),
+			DialContext:     CreateDialContext(t, ing, clients),
+			TLSClientConfig: tlsConfig,
 		},
 	}, cancel
 }
 
+// This is based on https://golang.org/src/crypto/tls/generate_cert.go
+func CreateTLSSecret(t *testing.T, clients *test.Clients, hosts []string) (string, context.CancelFunc) {
+	t.Helper()
+
+	priv, err := ecdsa.GenerateKey(elliptic.P256(), cryptorand.Reader)
+	if err != nil {
+		t.Fatalf("ecdsa.GenerateKey() = %v", err)
+	}
+
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := cryptorand.Int(cryptorand.Reader, serialNumberLimit)
+	if err != nil {
+		t.Fatalf("Failed to generate serial number: %v", err)
+	}
+
+	template := x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			Organization: []string{"Knative Ingress Conformance Testing"},
+		},
+
+		// Only let it live briefly.
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().Add(5 * time.Minute),
+
+		IsCA:                  true,
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+
+		DNSNames: hosts,
+	}
+
+	derBytes, err := x509.CreateCertificate(cryptorand.Reader, &template, &template, &priv.PublicKey, priv)
+	if err != nil {
+		t.Fatalf("x509.CreateCertificate() = %v", err)
+	}
+
+	cert, err := x509.ParseCertificate(derBytes)
```

Comment:
Perhaps we don't really need to validate that golang's x509 library works? :)

Reply:
not sure what you mean?

Reply:
nm.